scripts/cli.sh

```diff
@@ -0,0 +1,112 @@
+#! /bin/bash
+
+# A function to generate formatted .md files
+write_to_file() {
+    local cmd="$1"
+    local file="$2"
+    local program="$3"
+
+    # Remove first line of cmd to get rid of commit specific numbers.
+    cmd=${cmd#*$'\n'}
+
+    # We need to add the header and the backticks to create the code block.
+ printf "# %s\n\n\`\`\`\n%s\n\`\`\`" "$program" "$cmd" > "$file"
+}
+
+# Check if a lighthouse binary exists in the current branch.
+# -f means check if the file exists, to see all options, type "bash test" in a terminal
+maxperf=./target/maxperf/lighthouse
+release=./target/release/lighthouse
+debug=./target/debug/lighthouse
+
+if [[ -f "$maxperf" ]]; then
+    CMD="$maxperf"
+elif [[ -f "$release" ]]; then
+    CMD="$release"
+elif [[ -f "$debug" ]]; then
+    CMD="$debug"
+else
+    # No binary exists, build it.
+    cargo build --locked
+    CMD="$debug"
+fi
+
+# Store all help strings in variables.
+general_cli=$($CMD --help)
+bn_cli=$($CMD bn --help)
+vc_cli=$($CMD vc --help)
+am_cli=$($CMD am --help)
+vm_cli=$($CMD vm --help)
+
+general=./help_general.md
+bn=./help_bn.md
+vc=./help_vc.md
+am=./help_am.md
+vm=./help_vm.md
+
+# create .md files
+write_to_file "$general_cli" "$general" "Lighthouse General Commands"
+write_to_file "$bn_cli" "$bn" "Beacon Node"
+write_to_file "$vc_cli" "$vc" "Validator Client"
+write_to_file "$am_cli" "$am" "Account Manager"
+write_to_file "$vm_cli" "$vm" "Validator Manager"
+
+#input 1 = $1 = old files; input 2 = $2 = new files
+old_files=(./book/src/help_general.md ./book/src/help_bn.md ./book/src/help_vc.md ./book/src/help_am.md ./book/src/help_vm.md)
+new_files=($general $bn $vc $am $vm)
+
+# store for troubleshooting purpose
+exist=()
+update=() 
+diff=()
+
+# function to check
+check() {
```

Comment:
Personal preference, but I would create local variable here and assign them to the values of `$1` and `$2` (take a look at the `write_to_file` function for an example).

That way we can be more descriptive in the rest of the function ($1 is a bit opaque, imo).
I would do something like:

```
local file="$1"
local new_file="$2"
```

That way it's clear which file is which throughout the function.

Reply:
Local variables added in [0688991](https://github.com/sigp/lighthouse/pull/4571/commits/0688991f8761c0a81a7d4fed956e22d8a0c24ee7)